stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkPeerManager.java

```diff
@@ -50,16 +53,23 @@ public AsyncPrettyPrinterRegistry getAsyncPrettyPrinterRegistry() {
     return mAsyncPrettyPrinterRegistry;
   }
 
+  public void setPrettyPrinterRegister(@NonNull AsyncPrettyPrinterRegister register) {
+    mPrettyPrinterRegister = register;
+  }
+
   private final PeersRegisteredListener mTempFileCleanup = new PeersRegisteredListener() {
     @Override
     protected void onFirstPeerRegistered() {
+      AsyncPrettyPrinterExecutorHolder.init();
+      mAsyncPrettyPrinterRegistry = new AsyncPrettyPrinterRegistry();
+      mPrettyPrinterRegister.registerPrettyPrinters(mAsyncPrettyPrinterRegistry);
       mResponseBodyFileManager.cleanupFiles();
     }
 
     @Override
     protected void onLastPeerUnregistered() {
       mResponseBodyFileManager.cleanupFiles();
-      AsyncPrettyPrinterExecutorHolder.sExecutorService.shutdownNow();
+      AsyncPrettyPrinterExecutorHolder.getExecutorService().shutdownNow();
```

Comment:
This should be a first order function on `AsyncPrettyPrinterExecutorHolder` calling `shutdownNow` will not set the executor to null in `AsyncPrettyPrinterExecutorHolder`.